torch/nn/modules/rnn.py

```diff
@@ -66,6 +66,15 @@ def flatten_parameters(self):
             self._data_ptrs = []
             return
 
+        # If any parameters alias, we fall back to the slower, copying code path. This is
+        # a sufficient check, because overlapping parameter buffers that don't completely
+        # alias would break the assumptions of the uniqueness check in
+        # Module.named_parameters().
+        data_ptrs = list(p.data.data_ptr() for layer in self.all_weights for p in layer)
+        if len(set(data_ptrs)) != len(data_ptrs):
```

Comment:
It would be faster to construct it as a `set` and then compare to length of `self.all_weights`

Reply:
`self.all_weights` is nested, but I just moved the non-nested `all_weights` up from later in the code, so this should still be faster overall

Reply:
self.all_weights is nested, but I just changed to something that may be slightly faster?